extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -0,0 +1,642 @@
+  }
+
+
+  /**
+   * @param {boolean} isActive
+   * @public
+   */
+  setActive(isActive) {
+    if (isActive) {
+      this.pageActiveCallback_();
+    } else {
+      this.pageInactiveCallback_();
+    }
+  }
+
+
+  /** @private */
+  pageActiveCallback_() {
+    this.element.setAttribute('active', '');
+
+    if (!this.loadPromise_) {
+      return;
+    }
+
+    if (!this.loadTimeoutPromise_) {
+      this.loadTimeoutPromise_ = this.timer_.promise(LOAD_TIMEOUT_MS);
+    }
+
+    this.pageElements_.forEach(pageElement => {
+      pageElement.resumeCallback();
+    });
+
+    Promise.race([this.loadPromise_, this.loadTimeoutPromise_]).then(() => {
+      this.onPageVisible_();
+    });
+  }
+
+
+  /** @private */
+  pageInactiveCallback_() {
+    this.element.removeAttribute('active');
+
+    this.pauseAllMedia_();
+    this.pageElements_.forEach(pageElement => {
+      pageElement.pauseCallback();
+    });
+
+    if (this.autoAdvanceUnlistenDef_) {
+      this.autoAdvanceUnlistenDef_();
+      this.autoAdvanceUnlistenDef_ = null;
+    }
+
+    if (this.animationManager_) {
+      this.animationManager_.cancelAll();
+    }
+  }
+
+
+  /**
+   * @return {boolean} Whether this page is currently active.
+   * @public
+   */
+  isActive() {
+    return this.element.hasAttribute('active');
+  }
+
+
+  /**
+   * Gets an HTMLMediaElement from an element that wraps it.
+   * @param {!Element} el An element that wraps an HTMLMediaElement.
+   * @return {?Element} The underlying HTMLMediaElement, if one exists.
+   * @private
+   */
+  getMediaElement_(el) {
+    const tagName = el.tagName.toLowerCase();
+
+    if (el instanceof HTMLMediaElement) {
+      return el;
+    } else if (el.hasAttribute('background-audio') &&
+        (tagName === 'amp-story' || tagName === 'amp-story-page')) {
+      return el.querySelector('.i-amphtml-story-background-audio');
+    } else if (tagName === 'amp-audio') {
+      return el.querySelector('audio');
+    }
+
+    return null;
+  }
+
+
+  /**
+   * Determines whether a given element implements the video interface.
+   * @param {!Element} el The element to test
+   * @return {boolean} true, if the specified element implements the video
+   *     interface.
+   * @private
+   */
+  isVideoInterfaceVideo_(el) {
+    // TODO(newmuis): Check whether the element has the
+    // i-amphtml-video-interface class, after #11015 from amphtml is merged
+    // into amphtml-story.
+    const tagName = el.tagName.toLowerCase();
+
+    return tagName === 'amp-video' || tagName === 'amp-youtube' ||
+        tagName === 'amp-3q-player' || tagName === 'amp-ima-video' ||
+        tagName === 'amp-ooyala-player' || tagName === 'amp-nexxtv-player' ||
+        tagName === 'amp-brid-player' || tagName === 'amp-dailymotion';
+  }
+
+
+  /**
+   * Determines whether the specified element is a valid media element for auto-
+   * advance.
+   * @param {?Element} el
+   * @param {!function()} callback
+   * @return {!Element}
```

Comment:
This function doesn't seem to return anything.

Reply:
Done